mixed_requests/test_mixed.py

```diff
@@ -1,96 +1,379 @@
+"""
+
+
+
+class MixedRequests(tester.TempestaTest):
+
+    backends = [
+        {
+            'id' : 'nginx',
+            'type' : 'nginx',
+            'check_ports' : [
+                {
+                    "ip" : "${server_ip}",
+                    "port" : "8000",
+                }
+            ],
+            'status_uri' : 'http://${server_ip}:8000/nginx_status',
+            'config' : """
+pid ${backend_pid};
+worker_processes  auto;
+
+events {
+    worker_connections   1024;
+    use epoll;
+}
+
+http {
+    keepalive_timeout ${server_keepalive_timeout};
+    keepalive_requests ${server_keepalive_requests};
+    sendfile         on;
+    tcp_nopush       on;
+    tcp_nodelay      on;
+
+    open_file_cache max=1000;
+    open_file_cache_valid 30s;
+    open_file_cache_min_uses 2;
+    open_file_cache_errors off;
+
+    # [ debug | info | notice | warn | error | crit | alert | emerg ]
+    # Fully disable log errors.
+    error_log /dev/null emerg;
+
+    # Disable access log altogether.
+    access_log off;
+
+    server {
+        listen        ${server_ip}:8000;
+
+        location / {
+            return 200;
+        }
+        location /nginx_status {
+            stub_status on;
+        }
+    }
+}
+""",
+        }
+    ]
+
+    clients = [
+        {
+            'id' : 'wrk',
+            'type' : 'wrk',
+            'addr' : "${tempesta_ip}:80",
+        },
+    ]
+
+    tempesta = {
+        'config' : """
+cache 0;
+listen 80;
+
+srv_group default {
+    server ${server_ip}:8600;
+}
+
+vhost default {
+    proxy_pass default;
+}
+""",
+    }
+
+    def routine(self, lua):
+        nginx = self.get_server('nginx')
+        wrk = self.get_client("wrk")
+
+        wrk.set_script("wrk", lua)
+
+        nginx.start()
+        nginx.wait_for_connections(timeout=3)
+
+        self.start_tempesta()
+
+        wrk.start()
+        self.wait_while_busy(wrk)
+        wrk.stop()
+        return wrk
+
+    def test_pipeline(self):
+        self.routine(pipeline_lua)
+
+    def test_real(self):
+        self.routine(lua_real)
+
+    def test_real2(self):
+        self.routine(lua_real2)
+
+    def test_real_pipelined(self):
+        self.routine(lua_real_pipelined)
+
+    def test_get_post(self):
+        self.routine(lua_get_post)
+
+    def test_head_get(self):
+        self.routine(lua_head_get)
+
+    def test_post_empty(self):
+        self.routine(lua_post_empty)
+
+    def test_post_small(self):
+        self.routine(lua_post_small)
+
+    def test_post_big(self):
+        # Too big text to put it here explicitly
+
+        text = "content " * 8192
+        lua_post_big = r"""local body = [[
+""" + text + r"""]]
+
+wrk.method  = "POST"
+wrk.path = "/"
+wrk.headers = {["Content-Type"]="text/plain", ["Content-Length"]=string.len(body), ["Host"] = "localhost"}
+wrk.body    = body
+"""
+
+        self.routine(lua_post_big)
+
+    def test_mixed(self):
+        self.routine(lua_mixed)
+
+    # nginx always send 405 for TRACE
```

Comment:
We should provide such nginx config to make it sending 200 not 405. See the TRACE method description https://tools.ietf.org/html/rfc7231#section-4.3.8

Actually Trace method seems to be extremely handy for test frame work. Now you try to run wrk and deproxy.server in the same time to evaluate that Tempesta performs right under load. TRACE method gives and answer to you: what the final recipient (backend) has received. So curl +TRACE can replace that `deproxy.server` usage in stress tests.

**UPD.** As we discussed in the chat those 405 + connection closing is very specific situation. Actually not related to the TRACE method. This behaviour should be covered by it's own tests set. See https://github.com/tempesta-tech/tempesta/pull/1030#discussion_r192384448

Reply:
> We should provide such nginx config to make it sending 200 not 405. 

Why?

This tests checks that all work fine if client sends TRACE. 405 is a good answer in this case.

Reply:
Usually it's better to test only one case at one. 405 response and _connection close_ by backend is very specific case on its own, because in this case there is no requests pipelining in backend connections, a lot of 502 errors because of constantly unavailable connections. But the test intended to test responses to TRACE. That's why I commented the test,it seems that two different tests cases are checked in one test. 

I've double checked to Nginx and it seems there is no way to enable TRACE, it always responds this way:
https://github.com/nginx/nginx/blob/76be1ea9de13c5e8bb0d9523c6a2ad4009a5d7cf/src/http/ngx_http_request.c#L1825